### ISSUE TITLE
Consolidate labels on the repository

### ISSUE THREAD

**Issue by waiting-for-dev** [solidusio/solidus#4752]
In the process of automating our flow, we'll extensively leverage labels on issues and PRs.

After that, we need to immediately update the automatic labeler to use the new ones from the Changelog section below.

# Changelog

Usage: automatic generation of the release notes and the Changelog

Rename the following tags:

- Solidus Core → changelog:solidus_core
- Solidus Admin → changelog:solidus_admin
- Solidus API → changelog:solidus_api
- Solidus Frontend → changelog:solidus_frontend

Add the following tags:

- changelog:solidus_sample
- changelog:solidus (for changes that only affect the meta-gem)
- changelog:repository (for changes that affect files that don't belong to any gem)
- changelog:skip (explicitly marking something to not be in the Changelog)

# Backport

Usage: automatic backporting of PRs

Rename the following tags:

- Needs Backport -> backport
- Needs Backport-v3.1 -> backport:v3.1
- Needs Backport-v3.2 -> backport:v3.2

# Type

Usage: community support

Rename the following tags:

- Enhancement → type:enhancement
- Bug → type:bug

# Details

Usage: Details about a type

- Help wanted -> help wanted
- Confirmed → confirmed
- Security Concern → security
- Wontfix → wontfix
- Invalid → invalid

# Releases

Rename the following tags:

- Hold → release-major

Add the following tags:

- release-generate

# Other

Delete the following unused tags:

- Beginner-Friendly
- Challenge
- dependencies
- Discussion-General
- Discussion - Troubleshooting
- Documentation
- Good First Issue
- Hacktoberfest
- hacktoberfest-accepted
- I18n
- Important Change
- javascript
- JS
- Needs Core Team Decision
- Needs Core Team Review
- Needs Tests
- Needs Work
- Question
- Ruby
- SolidusConf Hack Day
- Stale
- System:Promotions
- UI
- Unconfirmed
- Waiting on Author Response
- WIP

**Comment by waiting-for-dev** [comment]
In the process of automating our flow, we'll extensively leverage labels on issues and PRs.

After that, we need to immediately update the automatic labeler to use the new ones from the Changelog section below.

# Changelog

Usage: automatic generation of the release notes and the Changelog

Rename the following tags:

- Solidus Core → changelog:solidus_core
- Solidus Admin → changelog:solidus_admin
- Solidus API → changelog:solidus_api
- Solidus Frontend → changelog:solidus_frontend

Add the following tags:

- changelog:solidus_sample
- changelog:solidus (for changes that only affect the meta-gem)
- changelog:repository (for changes that affect files that don't belong to any gem)
- changelog:skip (explicitly marking something to not be in the Changelog)

# Type

Usage: community support

Rename the following tags:

- Enhancement → type-enhancement
- Bug → type-bug
- Confirmed → confirmed
- Security Concern → security
- Wontfix → wontfix
- Invalid → invalid

# Releases

Rename the following tags:

- Hold → release-major

Add the following tags:

- release-generate

# Other

Delete the following unused tags:

- Beginner-Friendly
- Challenge
- dependencies
- Discussion-General
- Discussion - Troubleshooting
- Documentation
- Good First Issue
- Hacktoberfest
- hacktoberfest-accepted
- Help Wanted
- I18n
- Important Change
- javascript
- JS
- Needs Core Team Decision
- Needs Core Team Review
- Needs Tests
- Needs Work
- Question
- Ruby
- SolidusConf Hack Day
- Stale
- System:Promotions
- UI
- Unconfirmed
- Waiting on Author Response
- WIP

**Comment by gsmendoza** [comment]
@waiting-for-dev FYI, https://github.com/solidusio/.github/blob/main/.github/stale.yml depends on the `Stale` and `Enhancement` labels. Will you be updating the Stale config too?

**Comment by waiting-for-dev** [comment]
Thanks for looking into it, @gsmendoza. I had forgotten about updating the `Enhancement` label on the `stale.yml`. Now, both of them are changed in https://github.com/solidusio/.github/pull/3

**Comment by tvdeyen** [comment]
Why are we using a mixture of `_` `-` `:` to delimit the key from value in tags? Wouldn't it be more consistent to use only one (I would prefer `:` so we are still able to use `_` and `-` in the values)?

- `changelog:solidus_core`
- `backport:3.2`
- `type:bug`

**Comment by waiting-for-dev** [comment]
Sorry, they were typos. Everything is expected to use `:`. I updated the description.
